x-pack/test/reporting_api_integration/reporting_and_security/error_codes.ts

```diff
@@ -1,62 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-import expect from '@kbn/expect';
-import { INTERNAL_ROUTES } from '@kbn/reporting-common';
-import { ReportApiJSON } from '@kbn/reporting-common/types';
-import { FtrProviderContext } from '../ftr_provider_context';
-
-// eslint-disable-next-line import/no-default-export
-export default function ({ getService }: FtrProviderContext) {
-  const reportingAPI = getService('reportingAPI');
-  const esArchiver = getService('esArchiver');
-  const supertest = getService('supertestWithoutAuth');
-
-  describe('Reporting error codes', () => {
```

Comment:
The tests in this file are adequately provided in:
```
* x-pack/plugins/reporting/server/routes/internal/generate/integration_tests/generation_from_jobparams.test.ts
* x-pack/plugins/reporting/server/routes/public/integration_tests/generation_from_jobparams.test.ts
```

Reply:
nit: I looked at the test, and it does cover the test cases that exist in this file. However the test case and implementation for the returned error seems generic with a return message of `unknown_error`, is it worth considering returning an error that identifies the error as one that's terminally received from reporting?

Reply:
Hi @eokoneyo, in regards to this: I think that `unknown_error` you see is a subclass of a ReportingError, and if you view the entire error message, it should be distinguishable as an error from the reporting plugin. Would you mind double-checking or giving me more context to the `unknown_error` you are seeing?

Reply:
Slightly related to this topic, is that `unknown_error`, which is a subclass of ReportingError, is a catch-all for any errors we didn't specifically knew how to fingerprint in the original implementation. There may be patterns of `unknown_errors` in the wild, in which case we should make new subclasses of ReportingError to portray more types of error messages with more accurate meaning. Example: I think authentication errors might be getting mapped to `unknown_error`, but could be its own type of ReportingError.